torch/distributed/checkpoint/state_dict_saver.py

```diff
@@ -168,13 +178,14 @@ def _async_save(
 
     executor = ThreadPoolExecutor(max_workers=1)
     f = executor.submit(
-        _save_state_dict,
+        save,
```

Comment:
It is okay to use `save` because `_stateful_to_state_dict` will be an noop when it is called twice. The reason to call `save` is to ensure that we have the same way to handle the creation of StorageWriter for `async_save` and `save`.